### DISCUSSION TITLE
openapi vs. cypress

### DISCUSSION THREAD

**Discussion by jcszymansk** [realworld-apps/realworld#1249] [Q&A] [answered]
Hi,

I'm implementing a backend and I noticed that api/openapi.yml is apparently not exactly in sync with api cypress tests:

- yml doesn't list any error codes other than 401 and 422, tests expect also 403 and 404 in some places
- error json structure in the yml seems to me to be ```{"errors": {"body": ["an", "array", "of", "strings"]}}``` and this is what I got from the generator; this is not what tests expect, these seem to expect something like ```{"errors": "some object": ["is not what it should be"]}}```

I have so far adhered to the yml (I use openapi-generator to make res/resp object classes), but as most frontends seem to be on the cypress side I should perhaps switch sides as well, at least for error returning (I haven't so far noticed any differences wrt positive answers)?

**Comment by geromegrignon** [comment, marked as answer]
Hi @jacekszymanski, the Cypress API part is still a WIP I didn't include in the documentation yet.
You can stick with openapi.
